### PR TITLE
Refactor FilterExpressionCheck with entityClass

### PR DESCRIPTION
**Pull request by wcekan**
Update FilterExpressionCheck to match updated specification.  See https://github.com/yahoo/elide-doc/pull/37

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

**Comment by aklish** [comment]
👍

File: elide-datastore/elide-datastore-hibernate3/src/main/java/com/yahoo/elide/datastores/hibernate3/HibernateTransaction.java [hunk 1]
```diff
@@ -174,6 +174,10 @@ public void commit() {
     public <T> T loadObject(Class<T> loadClass, Serializable id, Optional<FilterExpression> filterExpression) {
 
         try {
+            // use load if no join or filter
+            if (!filterExpression.isPresent()) {
```
Comment (DeathByTape) [inline elide-datastore/elide-datastore-hibernate3/src/main/java/com/yahoo/elide/datastores/hibernate3/HibernateTransaction.java]: Does this need to be added to hibernate5 as well?

**Comment by DeathByTape** [comment]
👍

**Comment by aklish** [comment]
👍

**Pull request merged**